docs/design/2021-07-07-rule-based-index-selection.md

```diff
@@ -0,0 +1,74 @@
+# Proposal: Improve rule based index selection
+
+- Author(s): [xuyifangreeneyes](https://github.com/xuyifangreeneyes)
+- Last updated: 2021-08-13 
+- Discussion at:
```

Comment:
```suggestion
- Author(s): [xuyifangreeneyes](https://github.com/xuyifangreeneyes)
- Discussion PR: https://github.com/pingcap/tidb/pull/27223
- Tracking Issue: https://github.com/pingcap/tidb/issues/26020
```

Please follow the template https://raw.githubusercontent.com/pingcap/tidb/master/docs/design/TEMPLATE.md